From: Vincenzo Frascino <vincenzo.frascino@arm.com>
Subject: kasan: remove duplicate of kasan_flag_async

Patch series "arm64: ARMv8.7-A: MTE: Add asymm in-kernel support", v2.

This series implements the in-kernel asymmetric mode support for ARMv8.7-A
Memory Tagging Extension (MTE), which is a debugging feature that allows
to detect with the help of the architecture the C and C++ programmatic
memory errors like buffer overflow, use-after-free, use-after-return, etc.

MTE is built on top of the AArch64 v8.0 virtual address tagging TBI (Top
Byte Ignore) feature and allows a task to set a 4 bit tag on any subset of
its address space that is multiple of a 16 bytes granule.  MTE is based on
a lock-key mechanism where the lock is the tag associated to the physical
memory and the key is the tag associated to the virtual address.

When MTE is enabled and tags are set for ranges of address space of a
task, the PE will compare the tag related to the physical memory with the
tag related to the virtual address (tag check operation).  Access to the
memory is granted only if the two tags match.  In case of mismatch the PE
will raise an exception.

When asymmetric mode is present, the CPU triggers a fault on a tag
mismatch during a load operation and asynchronously updates a register
when a tag mismatch is detected during a store operation.

Note: The userspace support will be sent with a future patch series.


This patch (of 5):

After merging async mode for KASAN_HW_TAGS a duplicate of the
kasan_flag_async flag was left erroneously inside the code.

Remove the duplicate.

Note: This change does not bring functional changes to the code
base.

Link: https://lkml.kernel.org/r/20211004202253.27857-1-vincenzo.frascino@arm.com
Link: https://lkml.kernel.org/r/20211004202253.27857-2-vincenzo.frascino@arm.com
Signed-off-by: Vincenzo Frascino <vincenzo.frascino@arm.com>
Acked-by: Catalin Marinas <catalin.marinas@arm.com>
Reviewed-by: Andrey Konovalov <andreyknvl@gmail.com>
Cc: Dmitry Vyukov <dvyukov@google.com>
Cc: Andrey Ryabinin <aryabinin@virtuozzo.com>
Cc: Alexander Potapenko <glider@google.com>
Cc: Marco Elver <elver@google.com>
Cc: Evgenii Stepanov <eugenis@google.com>
Cc: Will Deacon <will@kernel.org>
Cc: Branislav Rankov <branislav.rankov@arm.com>
Cc: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
Cc: Suzuki K Poulose <suzuki.poulose@arm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/kasan/kasan.h |    2 --
 1 file changed, 2 deletions(-)

--- a/mm/kasan/kasan.h~kasan-remove-duplicate-of-kasan_flag_async
+++ a/mm/kasan/kasan.h
@@ -38,8 +38,6 @@ static inline bool kasan_async_mode_enab
 
 #endif
 
-extern bool kasan_flag_async __ro_after_init;
-
 #if defined(CONFIG_KASAN_GENERIC) || defined(CONFIG_KASAN_SW_TAGS)
 #define KASAN_GRANULE_SIZE	(1UL << KASAN_SHADOW_SCALE_SHIFT)
 #else
_
